ros/ros2/Utils.hpp

```diff
@@ -154,19 +158,20 @@ std::vector<Eigen::Vector3d> PointCloud2ToEigen(const PointCloud2 &msg) {
     return points;
 }
 
-PointCloud2 EigenToPointCloud2(const std::vector<Eigen::Vector3d> &points, const Header &header) {
-    PointCloud2 msg = CreatePointCloud2Msg(points.size(), header);
-    FillPointCloud2XYZ(points, msg);
-    return msg;
+PointCloud2::UniquePtr EigenToPointCloud2(const std::vector<Eigen::Vector3d> &points,
+                                          const Header &header) {
+    auto msg = CreatePointCloud2Msg(points.size(), header);
+    FillPointCloud2XYZ(points, *msg);
+    return std::move(msg);
```

Comment:
Do we actually need to cast to rvalue here? You are using it anyways when you call this utility function.

Reply:
Yes, it may be overkill (useless). In order to be sure, I've found this quick ref:
https://stackoverflow.com/questions/4316727/returning-unique-ptr-from-functions

Reply:
Ok, then I'm removing it (among with small changes).

I'll let you know when I finish so you give it a quick look ;)

Reply:
Nice :)